pkg/provider/ibmcloud/plugin/loadbalancer/lb.go

```diff
@@ -0,0 +1,466 @@
+			listenerFound = true
+			listenerToRemove := []string{*listener.Uuid}
+			attempt := 0
+			for {
+				attempt++
+				_, err = l.lbListenerService.DeleteLoadBalancerProtocols(&l.uuid, listenerToRemove)
+				if err == nil {
+					break
+				}
+				// Check for the lb is in state UPDATE_PENDING. (HTTP 500)
+				if strings.Contains(err.Error(), "UPDATE_PENDING") && attempt <= updateRetryCount {
+					time.Sleep(updateRetryTime)
+				} else {
+					return nil, err
+				}
+			}
+		}
+	}
+
+	// Listener was not found,so return an error
+	if listenerFound == false {
+		lbLogger.Error("unpublish", "name", l.name, "V", debugV, "listener not found", extPort)
+		return lbResult(""), fmt.Errorf("unknown port %v", extPort)
+	}
+
+	return lbResult("unpublish"), nil
+}
+
+// ConfigureHealthCheck configures the health checks for instance removal and reconfiguration
+// The parameters healthy and unhealthy indicate the number of consecutive success or fail pings required to
+// mark a backend instance as healthy or unhealthy. The ping occurs on the backendPort parameter and
+// at the interval specified.
+func (l *ibmcloudlb) ConfigureHealthCheck(hc loadbalancer.HealthCheck) (loadbalancer.Result, error) {
+	lbLogger.Debug("ConfigureHealthCheck", "name", l.name, "heathCheck", hc, "V", debugV)
+	l.lock.Lock()
+	defer l.lock.Unlock()
+
+	hmConfig := datatypes.Network_LBaaS_LoadBalancerHealthMonitorConfiguration{
+		BackendPort: &hc.BackendPort,
+		// IBM Clouud only has a single retry config, so make it the max of the two: healthy and unhealthy
+		MaxRetries: intPtr(max(hc.Healthy, hc.Unhealthy)),
+		Interval:   intPtr(int(hc.Interval / time.Second)),
+		Timeout:    intPtr(int(hc.Timeout / time.Second)),
+	}
+	hmArray := []datatypes.Network_LBaaS_LoadBalancerHealthMonitorConfiguration{hmConfig}
+
+	_, err := l.lbHealthMonitorService.UpdateLoadBalancerHealthMonitors(&l.uuid, hmArray)
+	if err != nil {
+		return nil, err
+	}
+
+	return lbResult("healthcheck"), nil
+}
+
+// RegisterBackend registers instances identified by the IDs to the LB's backend pool
+func (l *ibmcloudlb) RegisterBackends(ids []instance.ID) (loadbalancer.Result, error) {
+	lbLogger.Debug("RegisterBackends", "name", l.name, "ids", ids, "V", debugV)
+	l.lock.Lock()
+	defer l.lock.Unlock()
+
+	// Set default weight to 50 to match LB default
+	weight := 50
+
+	// Output array of members added
+	membersToAdd := []datatypes.Network_LBaaS_LoadBalancerServerInstanceInfo{}
+
+	// Get current list of backend members
+	lb, err := l.lbService.Mask("members").GetLoadBalancer(&l.uuid)
+	if err != nil {
+		return nil, err
+	}
+
+	// Iterate over the list of members to add checking if they already exist in the loadbalancer.
+	// If not, add member.
+	for _, id := range ids {
+		ip := string(id)
+		ipFound := false
+		for _, member := range lb.Members {
+			if *member.Address == ip {
+				ipFound = true
+				lbLogger.Error("RegisterBackends", "name", l.name, "V", debugV, "backend already exists", ip)
```

Comment:
Why is this an error? Won't this be hit if there is a rolling update and now we have a new member?

Reply:
I though I put this error message into the code pulling from either the simulator or AWS load balancer code, but can't find that now. This code logs an error message that the backend was already registered, but doesn't change any behavior. I'm fine with removing this check entirely, or downgrading the message to a warning/info/debug.